### PR TITLE
MOB-301 : update `connectNetwork` parameters to pass in memo field

### PR DESCRIPTION
**Pull request by mike-dydx**
[MOB-301 : update `connectNetwork` parameters to pass in memo field](https://linear.app/dydx/issue/MOB-301/update-connectnetwork-parameters-to-pass-in-memo-field)

Screenshot 2024-03-04 at 6 51 45 PM

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/state/manager/SystemUtils.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package exchange.dydx.abacus.state.manager
+
+enum class Platform(val rawValue: String) {
+    ios("ios"),
+    android("android"),
+    web("web"),
+}
+expect class SystemUtils {
```
Comment (mike-dydx) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/manager/SystemUtils.kt:8]: `expect` here is a way for Kotlin Multiplat to define a protocol which expects an `actual` implementation for each supported platform

File: src/iosMain/kotlin/exchange.dydx.abacus/state/manager/SystemUtils.ios.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package exchange.dydx.abacus.state.manager
+
+actual class SystemUtils {
```
Comment (mike-dydx) [inline src/iosMain/kotlin/exchange.dydx.abacus/state/manager/SystemUtils.ios.kt:3]: `actual` here is the "actualy" implementation of the `expect` protocol

File: src/nativeMain/AndroidManifest.xml [hunk 3]
```diff
@@ -1,2 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<manifest package="exchange.dydx.abacus.android" />
```
Comment (mike-dydx) [inline src/nativeMain/AndroidManifest.xml:2]: this was moved to `androidMain` instead of `nativeMain`

**Review by ruixhuang — APPROVED** [review]
👍

**Pull request merged**